src/services/messages-processor/service.ts

```diff
@@ -48,42 +54,59 @@ export const makeMessagesProcessor = ({
   s3Service: S3StoreService
   gsService: GsStoreService
 }) => {
-  const load = async () => {
+  const invalidExitMessageFiles = new Set<string>()
+
+  const loadNewMessages = async (messagesStorage: MessageStorage) => {
     if (!config.MESSAGES_LOCATION) {
       logger.debug('Skipping loading messages in webhook mode')
       return []
     }
 
-    logger.info(`Loading messages from ${config.MESSAGES_LOCATION}`)
+    logger.info(`Loading messages from '${config.MESSAGES_LOCATION}' folder`)
 
     const folder = await readFolder(config.MESSAGES_LOCATION)
 
-    const messages: ExitMessage[] = []
+    const messagesWithMetadata: ExitMessageWithMetadata[] = []
 
     logger.info('Parsing loaded messages')
 
-    for (const [ix, file] of folder.entries()) {
+    for (const [ix, messageFile] of folder.entries()) {
       logger.info(`${ix + 1}/${folder.length}`)
 
+      // skipping empty files
+      if (messageFile.content === '') {
+        logger.warn(`Empty file. Skipping...`)
+        invalidExitMessageFiles.add(messageFile.filename)
+        continue
+      }
+
+      // used for uniqueness of file contents
+      const fileChecksum = createHash('sha256')
+        .update(messageFile.content)
+        .digest('hex')
+
+      if (messagesStorage.touchMessageWithChecksum(fileChecksum)) {
+        logger.info(`File already loaded`)
+        continue
+      }
+
       let json: Record<string, unknown>
       try {
-        json = JSON.parse(file)
+        json = JSON.parse(messageFile.content)
```

Comment:
I wondered about the lifespan of messages. What if the service will run for a very long time, and the messages will only be added to our cache?
Should we add a check if the message was used to exit? And not include it in the cache?

Reply:
I don't think this can be done easily without rewriting major parts of Ejector